### PR TITLE
Add gradient bars for BarChart

### PR DESCRIPTION
**Pull request by larryonoff**
This PR adds functionality to draw gradient bars with `BarChartDataSet`

### CODE REVIEW

File: Source/Charts/Utils/ChartUtils.swift [hunk 1]
```diff
@@ -108,6 +108,14 @@ extension Double
 
 extension CGPoint
 {
+    static var infinite: CGPoint {
+        return CGPoint(x: CGFloat.infinity, y: CGFloat.infinity)
+    }
+
+    var isIntinite: Bool {
```
Comment (philfi) [inline Source/Charts/Utils/ChartUtils.swift]: `isInfinite` typo

Comment (larryonoff) [reply Source/Charts/Utils/ChartUtils.swift]: fixed and rebased on `branch/4.0.0`

**Comment by malikmani** [comment]
I need this functionality in the bar charts. Can you please add this in master branch.

File: ChartsDemo-iOS/Swift/Demos/BarChartViewController.swift [hunk 2]
```diff
@@ -118,28 +119,52 @@ class BarChartViewController: DemoBaseViewController {
...
         } else {
-            set1 = BarChartDataSet(values: yVals, label: "The year 2017")
-            set1.colors = ChartColorTemplates.material()
-            set1.drawValuesEnabled = false
-            
-            let data = BarChartData(dataSet: set1)
+            let set = BarChartDataSet(values: yVals, label: "The year 2017")
+            set.drawValuesEnabled = false
+
+            setup(set)
```
Comment (malikmani) [inline ChartsDemo-iOS/Swift/Demos/BarChartViewController.swift]: I think `setup(set)` can go out of this if-else condition.

**Comment by forbidden404** [comment]
Hello there,

I would appreciate this feature being introduced in the Framework. Is there any way I can help with the process? (e.g.: Continue developing the feature and fix merge issues, or maybe with CC and tests)

**Comment by larryonoff** [comment]
Nobody from the team responds on this PR for a long time. So I still use Charts from my branch

**Comment by liuxuan30** [comment]
sorry. but really busy for the team. I just saw my inbox there is another https://github.com/danielgindi/Charts/pull/3225 about bar gradient. would guys work together to see which is better?

**Comment by larryonoff** [comment]
I'm not sure, but this PR can be based on #3225. This was long time ago.

**Comment by liuxuan30** [comment]
if it's possible that you guys can work together to merge into one great pull request?

**Comment by pylapp** [comment]
Hi!

Could you be nice to process this pull request?
I may be cool to defined such background colors for data sets :)

Cheers

**Comment by liuxuan30** [comment]
I hope so,  but we need to decide which to merge, this or #3225.

**Comment by ChristianVinterly** [comment]
I think it's probably better to go with one and keep improving it than delay the decision. These 2 PRs have been ready for over a year now, and it would be great to get this functionality😊🙏

**Comment by liuxuan30** [comment]
if that is the case, than probably we need to take a look at both PR and decide the better one

**Comment by muinmomin** [comment]
@liuxuan30 Any idea which PR will be getting merged in? This one or #3225? Would love to have one of them merged in soon is possible. Seems like they both have passed all the checks but this one doesn’t have any pending changes. Thanks for your help!

**Comment by liuxuan30** [comment]
I will see if I can pull it off this or next week.

**Comment by liuxuan30** [comment]
first thing, this is targeted to 4.0 branch? I'd like it to be merged in master. any idea? @larryonoff

**Comment by larryonoff** [comment]
I believe that I rebased it properly https://github.com/danielgindi/Charts/pull/4411

**Comment by liuxuan30** [comment]
so we could close this and move to $4411?

**Comment by larryonoff** [comment]
@liuxuan30 lets close current PR

**Pull request closed without merging**